lib/OpenQA/WebAPI/Controller/API/V1/JobTemplate.pm

```diff
@@ -213,6 +213,88 @@ sub get_job_groups {
     return \%yaml;
 }
 
+sub create_or_update_job_template {
+    my ($self, $prio, $machine_name, $arch, $product_name, $product_spec, $job_template_name, $testsuite_name,
```

Comment:
Maybe just use `%args` here for named parameters? Performance-wise it will likely not make that much of a difference.